pilot/pkg/xds/ads.go

```diff
@@ -188,7 +188,8 @@ func (s *DiscoveryServer) receive(con *Connection, identities []string) {
 // handles 'push' requests and close - the code will eventually call the 'push' code, and it needs more mutex
 // protection. Original code avoided the mutexes by doing both 'push' and 'process requests' in same thread.
 func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *Connection) error {
-	if !s.shouldProcessRequest(con.proxy, req) {
+	if req.TypeUrl == v3.HealthInfoType {
```

Comment:
Handle HealthInfo explicitly to avoid unnecessary sotw->delta conversion for other request types